core/block/detailservice/set_details.go

```diff
@@ -0,0 +1,248 @@
+		return err
+	}
+	return s.objectLinksCollectionModify(spc.DerivedIDs().Archive, objectId, isArchived)
+}
+
+func (s *service) SetListIsFavorite(objectIds []string, isFavorite bool) error {
+	ids, err := s.store.HasIDs(objectIds...)
+	if err != nil {
+		return err
+	}
+	var (
+		anySucceed  bool
+		resultError error
+	)
+	for _, id := range ids {
+		err := s.SetIsFavorite(id, isFavorite)
+		if err != nil {
+			log.Error("failed to favorite object", zap.String("objectId", id), zap.Error(err))
+			resultError = errors.Join(resultError, err)
+		} else {
+			anySucceed = true
+		}
+	}
+	if resultError != nil {
+		log.Warn("failed to set objects as favorite", zap.Error(resultError))
+	}
+	if anySucceed {
+		return nil
+	}
+	return resultError
+}
+
+func (s *service) SetListIsArchived(objectIds []string, isArchived bool) error {
+	objectIDsPerSpace, err := s.partitionObjectIDsBySpaceID(objectIds)
+	if err != nil {
+		return fmt.Errorf("partition object ids by spaces: %w", err)
+	}
+
+	var (
+		multiErr   multierror.Error
+		anySucceed bool
+	)
+	for spaceID, objectIDs := range objectIDsPerSpace {
+		err = s.setIsArchivedForObjects(spaceID, objectIDs, isArchived)
+		if err != nil {
+			log.Error("failed to set isArchived to objects", zap.String("spaceId", spaceID),
+				zap.Strings("objectIDs", objectIDs), zap.Bool("isArchived", isArchived), zap.Error(err))
+			multiErr.Errors = append(multiErr.Errors, err)
+		} else {
+			anySucceed = true
+		}
+	}
+	if anySucceed {
+		return nil
+	}
+	return multiErr.ErrorOrNil()
+}
+
+func (s *service) checkArchivedRestriction(isArchived bool, objectId string) error {
+	if !isArchived {
+		return nil
+	}
+	return cache.Do(s, objectId, func(sb smartblock.SmartBlock) error {
+		return s.restriction.CheckRestrictions(sb, model.Restrictions_Delete)
+	})
+}
+
+func (s *service) objectLinksCollectionModify(collectionId string, objectId string, value bool) error {
+	return cache.Do(s, collectionId, func(b smartblock.SmartBlock) error {
+		coll, ok := b.(collection.Collection)
+		if !ok {
+			return fmt.Errorf("unsupported sb block type: %T", b)
+		}
+		if value {
+			return coll.AddObject(objectId)
+		} else {
+			return coll.RemoveObject(objectId)
+		}
+	})
+}
+
+func (s *service) partitionObjectIDsBySpaceID(objectIDs []string) (map[string][]string, error) {
+	res := map[string][]string{}
+	for _, objectID := range objectIDs {
+		spaceID, err := s.resolver.ResolveSpaceID(objectID)
+		if err != nil {
+			return nil, fmt.Errorf("resolve spaceID: %w", err)
+		}
+		res[spaceID] = append(res[spaceID], objectID)
+	}
+	return res, nil
+}
+
+func (s *service) setIsArchivedForObjects(spaceID string, objectIDs []string, isArchived bool) error {
+	spc, err := s.spaceService.Get(context.Background(), spaceID)
+	if err != nil {
+		return fmt.Errorf("get space: %w", err)
+	}
+	return cache.Do(s, spc.DerivedIDs().Archive, func(b smartblock.SmartBlock) error {
+		archive, ok := b.(collection.Collection)
```

Comment:
Maybe extract to separate function?

Reply:
I have splitted this function onto two separate ones